server.js

```diff
@@ -0,0 +1,35 @@
+"use strict";
+
+var express = require('express');
+var app = express();
+var server = require('http').createServer(app);
+var Grid = require('./src/grid.js');
+var Cell = require('./src/cell.js');
+
+app.use(express.static(__dirname + '/public'));
+
+var grid;
+
+app.get('/', function (req, res) {
+  grid = new Grid(Cell)
+  grid.init();
+  res.sendFile(__dirname + '/views/index.html');
+});
+
+app.post('/', function (req, res) {
+  var buf = "";
+  req.on('data', function(chunk) {
+    buf += chunk.toString();
+  })
+  .on('end', function() {
+    grid.loadPuzzle(buf.split("=")[1]);
```

Comment:
It's not clear what you're doing with buf.split, you should use the body parser node module.